Mention Pod.spec.enableServiceLinks influence on container environment variables (#28042)
kubernetes/website: content/en/docs/concepts/containers/container-environment.md

### Cluster information

> When the [Pod spec](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#pod-v1-core) field `enableServiceLinks` is `true` (the default),

Reviewer: Hi. I would actually link to https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#containers - it's a newer API reference.

> a list of services are available to the Pod's Containers as environment variables.

Reviewer: ```suggestion
a list of Services are available to the Pod's containers as environment variables.
```

> This list is limited to services within the same namespace as the new Container's Pod and Kubernetes control plane services, which existed at the time the Container was created.

Reviewer: Is this also correct?
```suggestion
This list is limited to:

- Services within the same namespace as the Pod 
- Services representing the Kubernetes control plane, providing that they existed at the time the Pod was created.
```

> These environment variables match the syntax of Docker links.

Reviewer: Let's explain this syntax. We can't assume that readers are familiar with Docker.

```

Services have dedicated IP addresses and are available to the Container via DNS,
if [DNS addon](https://releases.k8s.io/{{< param "githubbranch" >}}/cluster/addons/dns/) is enabled.

> Read [Connecting Applications with Services](/docs/concepts/services-networking/connect-applications-service/).

Reviewer: I'd move this into the What's Next section.
